sql-plugin/src/main/scala/com/nvidia/spark/rapids/conditionalExpressions.scala

```diff
@@ -25,36 +25,8 @@ import org.apache.spark.sql.catalyst.expressions.{ComplexTypeMergingExpression,
 import org.apache.spark.sql.types.{BooleanType, DataType, DataTypes}
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
-trait GpuConditionalExpression extends ComplexTypeMergingExpression with GpuExpression
-  with ShimExpression {
-
-  protected def computeIfElse(
-      batch: ColumnarBatch,
-      predExpr: Expression,
-      trueExpr: Expression,
-      falseValue: Any): GpuColumnVector = {
-    withResourceIfAllowed(falseValue) { falseRet =>
-      withResource(GpuExpressionsUtils.columnarEvalToColumn(predExpr, batch)) { pred =>
-        withResourceIfAllowed(trueExpr.columnarEval(batch)) { trueRet =>
-          val finalRet = (trueRet, falseRet) match {
-            case (t: GpuColumnVector, f: GpuColumnVector) =>
-              pred.getBase.ifElse(t.getBase, f.getBase)
-            case (t: GpuScalar, f: GpuColumnVector) =>
-              pred.getBase.ifElse(t.getBase, f.getBase)
-            case (t: GpuColumnVector, f: GpuScalar) =>
-              pred.getBase.ifElse(t.getBase, f.getBase)
-            case (t: GpuScalar, f: GpuScalar) =>
-              pred.getBase.ifElse(t.getBase, f.getBase)
-            case (t, f) =>
-              throw new IllegalStateException(s"Unexpected inputs" +
-                s" ($t: ${t.getClass}, $f: ${f.getClass})")
-          }
-          GpuColumnVector.from(finalRet, dataType)
-        }
-      }
-    }
-  }
 
+trait GpuColumnVectorHelper extends Arm {
```

Comment:
Can we find a better name for this? It was `GpuConditionalExpression` so it was clear that it was supported to be for use with those, but now it is a generic utility library with an even more generic name and no clear contract on how these APIs should be used.  Could we change this to be an object instead of a trait? Could we restore the name of `GpuConditionalExpression` so it is still clear that they are intended to be used with these types of classes? Also I really dislike that we are overriding `isAllTrue` that just does not appear to make and since to me.  If we have different ways of calculating `isAllTrue` then we should either have separate utility APIs with a name that makes it clear how each works, or we should have a flag passed to the API to allow you to select how it works.  Does that make since to you?

Reply:
Yes, that makes sense.
I initially thought to use the existing Object `GpuExpressionsUtils`. Should that be fine? Or do you prefer to create a new Object helper , and what would be its name?